test/system/src/test/java/io/pravega/test/system/ControllerFailoverTest.java

```diff
@@ -63,34 +66,49 @@ public static void setup() throws InterruptedException, MarathonException, URISy
         //get the zk ip details and pass it to bk, host, controller
         URI zkUri = zkUris.get(0);
         //2, check if bk is running, otherwise start, get the zk ip
-        Service bkService = new BookkeeperService("bookkeeper", zkUri);
+        Service bkService = Utils.createBookkeeperService(zkUri);
         if (!bkService.isRunning()) {
             bkService.start(true);
         }
 
         List<URI> bkUris = bkService.getServiceDetails();
         log.debug("bookkeeper service details: {}", bkUris);
 
+        //start HDFS
+        URI hdfsUri = null;
+        if (Utils.isDockerLocalExecEnabled()) {
```

Comment:
Same comment about environment specific behavior on test cases.

Reply:
answered above.  need to deploy hdfs as it will not come as a part of cluster.